pkg/profile/artifact.go

```diff
@@ -90,26 +80,51 @@ func (pa *ProfilesArtifactsMaker) Make(installation profilesv1.ProfileInstallati
 			}
 		}
 		for _, obj := range artifact.Objects {
-			filename := filepath.Join(artifactDir, fmt.Sprintf("%s.%s", obj.GetObjectKind().GroupVersionKind().Kind, "yaml"))
-			if err := pa.generateOutput(filename, obj); err != nil {
+			name := obj.GetObjectKind().GroupVersionKind().Kind
+			if obj.Name != "" {
+				name = obj.Name
+			}
+			filename := filepath.Join(artifactDir, fmt.Sprintf("%s.%s", name, "yaml"))
+			if obj.Path != "" {
+				subFolder := filepath.Join(artifactDir, obj.Path)
+				if err := os.MkdirAll(subFolder, 0755); err != nil && !os.IsExist(err) {
+					return fmt.Errorf("failed to create directory")
+				}
+				filename = filepath.Join(subFolder, fmt.Sprintf("%s.%s", name, "yaml"))
+			}
+			if err := pa.generateOutput(filename, obj.Object); err != nil {
 				return err
 			}
 		}
+		// if we have a local resource, write out the kustomization yaml limiting its visibility.
 		if artifact.Kustomize != nil {
```

Comment:
is it ever not set?

Reply:
Yes. This probably comes from the fact that it's named poorly. :D I'll try and rename it to give it better context.